src/neo/SmartContract/ContractParametersContext.cs

```diff
@@ -144,16 +144,16 @@ public bool AddSignature(Contract contract, ECPoint pubkey, byte[] signature)
                     int i = 0;
                     switch (contract.Script[i++])
                     {
-                        case 1:
+                        case (byte)OpCode.PUSHINT8:
                             ++i;
                             break;
-                        case 2:
+                        case (byte)OpCode.PUSHINT16:
                             i += 2;
                             break;
                     }
-                    while (contract.Script[i++] == 33)
+                    while (contract.Script[i++] == (byte)OpCode.PUSHDATA1)
```

Comment:
```suggestion
                    while (contract.Script[i++] == (byte)OpCode.PUSHDATA1) && contract.Script[i++] == 33
```

Reply:
It is checked in the next line.

Reply:
We take 33, but it could be more than this

Reply:
What do you mean? I have test this method.

Reply:
The problem is that we don't  check that the next byte is exactly 33

Reply:
It has been checked here:

https://github.com/neo-project/neo/blob/8d2b0ba79ce6833177ae898ba00c6990facfe93c/src/neo/SmartContract/ContractParametersContext.cs#L133